### PR TITLE
[Lesson 6: Task Manager API Part 2] Code Assignment

### PR DESCRIPTION
**Pull request by yuliahar**
1. Task: https://learn.codethedream.org/ctd-node-lesson6-fall2023/

2. Coding Assignment:
3. 
- [x] Add validations and middleware to handle errors and asyncWrapper.
- [x] Quiz

4. Done 10.31.2023 / deadline 11.07.2023

### CODE REVIEW

File: 03-task-manager/starter/models/Task.js
Comment (leo-mutuku) [inline 03-task-manager/starter/models/Task.js:1]: task model well defined with correct constraints

File: 03-task-manager/starter/controllers/tasks.js
Comment (leo-mutuku) [inline 03-task-manager/starter/controllers/tasks.js:1]: Task CRUDE well implimented, http method (post, get, update, delete)

**Review by leo-mutuku — APPROVED** [review]
Tassk controller and model well implimented

File: 03-task-manager/starter/quizAnswers2.txt
Comment (leo-mutuku) [inline 03-task-manager/starter/quizAnswers2.txt:1]: asyn wrapper middleware well created, customer Error class not implimented

**Review by leo-mutuku — APPROVED** [review]
Tassk controller and model well implimented

File: 03-task-manager/starter/quizAnswers2.txt
Comment (yuliahar) [reply 03-task-manager/starter/quizAnswers2.txt:1]: Hello, thank you for your review! I completely forgot to include the CustomAPIError class :) I appreciate that you pointed that out. I'm providing the code for this class below:

class CustomAPIError extends Error {
  constructor(message, statusCode) {
    super(message);
    this.statusCode = statusCode;
  }
}

const createCustomError = (message, statusCode) => {
  return new CustomAPIError(message, statusCode);
};

module.exports = { CustomAPIError, createCustomError };

**Pull request merged**